clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -65,6 +65,9 @@
     /* The v2 metadata request is the same as v1. An additional field for cluster id has been added to the v2 metadata response */
     public static final Schema METADATA_REQUEST_V2 = METADATA_REQUEST_V1;
 
+    /* The v3 metadata request is the same as v1 and v2. An additional field for throttle time has been added to the v3 metadata response */
```

Comment:
Sorry for the delay on this one. So, one way to do this would be the following:

1. Add two fields to ResponseHeader _if_ the request version is higher than the version before this PR: error_code, throttle_time_ms (in this order)
2. Remove any top-level error_code in the new version of all affected responses
3. Remove throttle_time_ms from FetchResponse and ProduceResponse in the new version
4. throttle_time_ms is always 0 for requests that are never throttled

As part of this, we would also solve the issue that we currently have no way to return generic errors via the protocol. Since we are bumping the protocol version for so many requests, it seems like it would be a good opportunity to fix both issues at the same time. Is there a reason why this is a bad idea?